packages/cinterop/src/darwin/kotlin/io/realm/kotlin/internal/interop/RealmInterop.kt

```diff
@@ -1879,8 +1882,19 @@ actual object RealmInterop {
             staticCFunction { userData, beforeRealm, afterRealm, didRecover ->
                 val afterCallback = safeUserData<SyncAfterClientResetHandler>(userData)
                 val beforeDb = CPointerWrapper<FrozenRealmT>(beforeRealm, false)
-                val afterDb = CPointerWrapper<LiveRealmT>(afterRealm, false)
-                afterCallback.onAfterReset(beforeDb, afterDb, didRecover)
+
+                // afterRealm is wrapped inside a ThreadSafeReference so the pointer needs to be resolved
+                val afterRealmPtr = realm_wrapper.realm_from_thread_safe_reference(afterRealm, null)
+                val afterDb = CPointerWrapper<LiveRealmT>(afterRealmPtr, false)
+
+                // Check if exceptions have been thrown, return true if all went as it should
+                try {
+                    afterCallback.onAfterReset(beforeDb, afterDb, didRecover)
+                    true
+                } catch (e: Throwable) {
+                    println(e.message)
```

Comment:
Shouldn't we use our internal logger so this message could be captured by a custom logger?

Maybe the exception catch could happen in a higher layer.

Reply:
The internal logger isn't available in cinterop unfortunately. I'm not sure there is good way to get this to work right now, but it is something we should track and hopefully fix when we look into combining our modules.